torch/csrc/distributed/autograd/context/context.cpp

```diff
@@ -193,6 +202,28 @@ std::shared_ptr<rpc::FutureMessage> DistAutogradContext::
   return state->future;
 }
 
+void DistAutogradContext::setAutogradError(const std::string& errorMsg) {
+  setGraphTaskException(errorMsg);
+  propagateAutogradError(errorMsg);
+}
```

Comment:
should be done in request_callback_impl.cpp by just retrieving information from the context.